src/Compilers/Core/Portable/PEWriter/TypeNameSerializer.cs

```diff
@@ -88,6 +88,25 @@ internal static string GetSerializedTypeName(this ITypeReference typeReference,
                     goto done;
                 }
 
+                if (allowTypeParameters)
+                {
+                    // When encoding types into extension erasure attributes we allow type parameters (encoded as `!N` and `!!N`).
+                    if (typeReference.AsGenericTypeParameterReference is { } genericTypeParameter)
+                    {
+                        sb.Append("!");
+                        var index = MetadataWriter.GetNumberOfInheritedTypeParameters(genericTypeParameter.DefiningType, context) + genericTypeParameter.Index;
+                        sb.Append(index);
```

Comment:
>Append(index) [](http://example.com/codeflow?start=27&length=13)

Is there a way to specify specific format here? I would be good to avoid dependency on default format which could be affected by current locale, etc.